src/scportrait/pipeline/_utils/sdata_io.py

```diff
@@ -160,174 +219,73 @@ def _write_points_object_sdata(self, points, points_name: str, overwrite: bool =
+        """Add cell centers from segmentation.
+
+        Args:
+            segmentation_label: Label of segmentation to use
+            overwrite: Whether to overwrite existing centers
+        """
         _sdata = self._read_sdata()
         centroids_object = self._get_centers(_sdata, segmentation_label)
         self._write_points_object_sdata(centroids_object, self.centers_name, overwrite=overwrite)
 
-    def _load_input_image_to_memmap(self, tmp_dir_abs_path: str, image=None):
-        """
-        Helper function to load the input image from sdata to memory mapped temp arrays for faster access.
-        Loading happens in a chunked manner to avoid memory issues.
+    def _load_input_image_to_memmap(self, tmp_dir_abs_path: str, image: xarray.DataArray | None = None) -> str:
+        """Load input image to memory mapped array.
 
-        The function will return the path to the memory mapped array.
+        Args:
+            tmp_dir_abs_path: Path for temporary storage
+            image: Optional image data to load
 
-        Parameters
-        ----------
-        tmp_dir_abs_path : str
-            Absolute path to the directory where the memory mapped arrays should be stored.
+        Returns:
+            Path to memory mapped array
 
-        Returns
-        -------
-        str
-            Path to the memory mapped array. Can be reconneted to using the `mmap_array_from_path`
-            function from the alphabase.io.tempmmap module.
+        Raises:
+            ValueError: If input image not found
         """
-        # ensure all elements are loaded
         if image is None:
             _sdata = self._check_sdata_status(return_sdata=True)
-
             if not self.input_image_status:
                 raise ValueError("Input image not found in sdata object.")
-
             image = self._get_input_image(_sdata)
-        shape = image.shape
 
-        # initialize empty memory mapped arrays to store the data
+        shape = image.shape
         path_input_image = tempmmap.create_empty_mmap(
             shape=shape,
             dtype=image.dtype,
             tmp_dir_abs_path=tmp_dir_abs_path,
         )
 
-        # create the empty mmap array
         input_image_mmap = tempmmap.mmap_array_from_path(path_input_image)
 
-        # load the data into the mmap array in chunks
-        Z = None
         if len(shape) == 3:
             C, Y, X = shape
-
+            for c in range(C):
+                input_image_mmap[c] = image[c].compute()
         elif len(shape) == 4:
             Z, C, Y, X = shape
-
-        if Z is not None:
             for z in range(Z):
                 for c in range(C):
                     input_image_mmap[z][c] = image[z][c].compute()
-        else:
-            for c in range(C):
-                input_image_mmap[c] = image[c].compute()
 
-        # cleanup the cache
         del input_image_mmap, image
-
         return path_input_image
-
```

Comment:
specific reason this function was removed?

Reply:
Apologies - should be back now.